src/operations/blas2/tbmv.hpp

```diff
@@ -0,0 +1,132 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename tbmv.hpp
+ *
+ **************************************************************************/
+
+#ifndef TBMV_HPP
+#define TBMV_HPP
+#include "operations/blas2_trees.h"
+#include "operations/blas_operators.hpp"
+#include "views/view_sycl.hpp"
+#include <stdexcept>
+#include <vector>
+namespace blas {
+
+/**
+ * @struct Tbmv
+ * @brief Tree node representing a triangular band matrix_ vector_
+ * multiplication.
+ */
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_upper, bool is_transposed,
+          bool is_unitdiag>
+SYCL_BLAS_INLINE
+Tbmv<lhs_t, matrix_t, vector_t, local_range, is_upper, is_transposed,
+     is_unitdiag>::Tbmv(lhs_t &_l, matrix_t &_matrix,
+                        typename Tbmv<lhs_t, matrix_t, vector_t, local_range,
+                                      is_upper, is_transposed,
+                                      is_unitdiag>::index_t &_k,
+                        vector_t &_vector)
+    : lhs_(_l), matrix_(_matrix), vector_(_vector), k_(_k) {}
+
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_upper, bool is_transposed,
+          bool is_unitdiag>
+SYCL_BLAS_INLINE typename Tbmv<lhs_t, matrix_t, vector_t, local_range, is_upper,
+                               is_transposed, is_unitdiag>::index_t
+Tbmv<lhs_t, matrix_t, vector_t, local_range, is_upper, is_transposed,
+     is_unitdiag>::get_size() const {
+  return matrix_.get_size();
+}
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_upper, bool is_transposed,
+          bool is_unitdiag>
+SYCL_BLAS_INLINE bool
+Tbmv<lhs_t, matrix_t, vector_t, local_range, is_upper, is_transposed,
+     is_unitdiag>::valid_thread(cl::sycl::nd_item<1> ndItem) const {
+  // Valid threads are established by ::eval.
+  return true;
+}
+
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_upper, bool is_transposed,
+          bool is_unitdiag>
+SYCL_BLAS_INLINE typename Tbmv<lhs_t, matrix_t, vector_t, local_range, is_upper,
+                               is_transposed, is_unitdiag>::value_t
+Tbmv<lhs_t, matrix_t, vector_t, local_range, is_upper, is_transposed,
+     is_unitdiag>::eval(cl::sycl::nd_item<1> ndItem) {
+  const index_t lhs_idx =
+      ndItem.get_group(0) * local_range + ndItem.get_local_id(0);
```

Comment:
I think you can use  `lhs_idx= ndItem.get_global(0);` instead and avoid this computation here.

Reply:
Done!